datagateway_api/src/search_api/query_filter_factory.py

```diff
@@ -0,0 +1,151 @@
+            conditions = list(filter_input.values())[0]
+            for condition in conditions:
+                # Could be nested AND/OR
+                filter_data = SearchAPIQueryFilterFactory.get_condition_values(
+                    condition,
+                )
+                for condition in filter_data:
+                    where_filters.append(
+                        SearchAPIWhereFilter(
+                            field=condition[0],
+                            value=condition[1],
+                            operation=condition[2],
+                            boolean_operator=boolean_operator,
+                        ),
+                    )
+        else:
+            filter_data = SearchAPIQueryFilterFactory.get_condition_values(
+                filter_input,
+            )
+            for condition in filter_data:
+                where_filters.append(
+                    SearchAPIWhereFilter(
+                        field=condition[0], value=condition[1], operation=condition[2],
+                    ),
+                )
+        return where_filters
+
+    @staticmethod
+    def get_include_filter(filter_input):
+        query_filters = []
+        for related_model in filter_input:
+            included_entity = related_model["relation"]
+            query_filters.append(SearchAPIIncludeFilter(included_entity))
+
+            if "scope" in related_model:
+                try:
+                    where_key = list(related_model["scope"]["where"].keys(),)[0]
+
+                    if where_key == "text":
+                        # TODO - we might want to move this to the data
+                        # definitions at a later point
+                        text_operator_fields = {
+                            "datasets": ["title"],
+                            "documents": ["title", "summary"],
+                            "files": ["name"],
+                            "instrument": ["name", "facility"],
+                            "samples": ["name", "description"],
+                            "techniques": ["name"],
+                        }
+                        field_names = text_operator_fields[included_entity]
+                    else:
+                        field_names = [where_key]
+
+                    for field_name in field_names:
+                        full_field_path = f"{included_entity}.{field_name}"
+                        where_filter = {
+                            "filter": {
+                                "where": {
+                                    full_field_path: related_model["scope"]["where"][
+                                        where_key
+                                    ],
+                                },
+                            },
+                        }
+                        query_filters.extend(
+                            SearchAPIQueryFilterFactory.get_query_filter(where_filter,),
+                        )
```

Comment:
@MRichards99 Correct me if I am wrong but my understanding here is that two `SearchAPIWhereFilter` objects will be created, one after another, if there are two elements in `field_names`. However, wouldn't this mean that both of these filter objects will need to find a match for the provided value? 

For example, imagine we were querying the `/Datasets` endpoint with a relation `instrument` and _WHERE_ filter `"text": "ESS" `. The text operator fields for `instrument` are `name` and `facility`, meaning that there will be one WHERE filter for `instrument.name` and another for `instrument.facility`. Wouldn't this mean that it will look for `instrument.name` with value `ESS` **AND** `instrument.facility` with value `ESS`?

Shouldn't it be instead looking for `instrument.name` with value `ESS` **OR** `instrument.facility` with value `ESS`? If so then I am guessing we can create an _OR WHERE_ filter here and send that to `get_query_filter`?

Reply:
I agree, the text operator should create where filters using the OR boolean operator. Good spot